internal/driver/onvifdiscovery_test.go

```diff
@@ -0,0 +1,349 @@
+					EndpointRefAddress: "123",
+				},
+			},
+		},
+		{
+			Name: "testDevice2", Protocols: map[string]models.ProtocolProperties{
+				OnvifProtocol: map[string]string{
+					EndpointRefAddress: "456",
+				},
+			},
+		},
+		{
+			Name: "testDevice3", Protocols: map[string]models.ProtocolProperties{
+				OnvifProtocol: map[string]string{
+					EndpointRefAddress: "789",
+				},
+			},
+		},
+	}
+}
+
+func TestOnvifDiscovery_makeDeviceMap(t *testing.T) {
+	tests := []struct {
+		name      string
+		devices   []contract.Device
+		deviceMap map[string]contract.Device
+		nameCalls int
+	}{
+		{
+			name:    "3 devices",
+			devices: createTestDeviceList(),
+			deviceMap: map[string]contract.Device{
+				"123": {
+					Name: "testDevice1", Protocols: map[string]models.ProtocolProperties{
+						OnvifProtocol: map[string]string{
+							EndpointRefAddress: "123",
+						},
+					},
+				},
+				"456": {
+					Name: "testDevice2", Protocols: map[string]models.ProtocolProperties{
+						OnvifProtocol: map[string]string{
+							EndpointRefAddress: "456",
+						},
+					},
+				},
+				"789": {
+					Name: "testDevice3", Protocols: map[string]models.ProtocolProperties{
+						OnvifProtocol: map[string]string{
+							EndpointRefAddress: "789",
+						},
+					},
+				},
+			},
+			nameCalls: 4,
+		},
+		{
+			name: "NoProtocol",
+			devices: []contract.Device{
+				{
+					Name: "testDevice1",
+					Protocols: map[string]models.ProtocolProperties{
+						OnvifProtocol: map[string]string{
+							EndpointRefAddress: "123",
+						},
+					},
+				},
+				{
+					Name:      "testDevice2",
+					Protocols: map[string]models.ProtocolProperties{},
+				},
+			},
+			deviceMap: map[string]contract.Device{
+				"123": {
+					Name: "testDevice1", Protocols: map[string]models.ProtocolProperties{
+						OnvifProtocol: map[string]string{
+							EndpointRefAddress: "123",
+						},
+					},
+				},
+			},
+			nameCalls: 2,
+		},
+		{
+			name: "NoEndpointReference",
+			devices: []contract.Device{
+				{
+					Name: "testDevice1",
+					Protocols: map[string]models.ProtocolProperties{
+						OnvifProtocol: map[string]string{
+							EndpointRefAddress: "123",
+						},
+					},
+				},
+				{
+					Name: "testDevice2",
+					Protocols: map[string]models.ProtocolProperties{
+						OnvifProtocol: map[string]string{
+							EndpointRefAddress: "",
+						},
+					},
+				},
+			},
+			deviceMap: map[string]contract.Device{
+				"123": {
+					Name: "testDevice1", Protocols: map[string]models.ProtocolProperties{
+						OnvifProtocol: map[string]string{
+							EndpointRefAddress: "123",
+						},
+					},
+				},
+			},
+			nameCalls: 2,
+		},
+	}
+	for _, test := range tests {
+		test := test
+		t.Run(test.name, func(t *testing.T) {
+			driver, mockService := createDriverWithMockService()
+			mockService.On("Devices").
+				Return(test.devices).Once()
+			mockService.On("Name").
+				Return("device-onvif-camera").Times(test.nameCalls)
```

Comment:
i think the amount of times this is actually called we do not care about. is there a way to ignore that?

Reply:
yes, if you leave off the "Times" part, and do not use any of the `Once`, `Twice`, etc. it will ignore the amount of times it gets called.